src/tests/functional/plugin/cpu/single_layer_tests/gather.cpp

```diff
@@ -1,345 +1,343 @@
+            inputDynamicShapes.push_back({1});
+            for (size_t i = 0lu; i < targetStaticShapes.size(); i++) {
+                targetStaticShapes[i].push_back({1});
+            }
+        }
+
+        ngraph::ParameterVector params {
+            std::make_shared<ov::op::v0::Parameter>(netPrecision, inputDynamicShapes[0]),
+            std::make_shared<ov::op::v0::Parameter>(intInputsPrecision, inputDynamicShapes[1])
+        };
+        params[0]->set_friendly_name("data");
+        params[1]->set_friendly_name("indices");
+        if (!isAxisConstant) {
+            params.push_back(std::make_shared<ov::op::v0::Parameter>(intInputsPrecision, inputDynamicShapes[2]));
+            params[2]->set_friendly_name("axis");
+        }
+        auto paramOuts = ngraph::helpers::convert2OutputVector(ngraph::helpers::castOps2Nodes<ov::op::v0::Parameter>(params));
+        std::shared_ptr<ov::Node> gatherNode;
+        if (isAxisConstant) {
+            gatherNode = std::make_shared<ov::op::v8::Gather>(paramOuts[0], paramOuts[1],
+                    ov::op::v0::Constant::create(intInputsPrecision, ov::Shape({1}), { axis }), batchDims);
+        } else {
+            gatherNode = std::make_shared<ov::op::v8::Gather>(paramOuts[0], paramOuts[1], paramOuts[2], batchDims);
+        }
+
+        function = makeNgraphFunction(netPrecision, params, gatherNode, "GatherCPU");
+    }
+
+    void generate_inputs(const std::vector<ov::Shape>& targetInputStaticShapes) override {
+        const auto& funcInputs = function->inputs();
+        inputs.clear();
+
+        const size_t normAxis = axis < 0 ? axis + targetInputStaticShapes[0].size() : axis;
+        const int32_t axisDim = targetInputStaticShapes[0][normAxis];
+
+        for (int i = 0; i < funcInputs.size(); ++i) {
+            const auto& funcInput = funcInputs[i];
+            ov::runtime::Tensor tensor;
+
+            if (funcInput.get_node()->get_friendly_name() == "data") {
+                tensor = ov::test::utils::create_and_fill_tensor(
+                        funcInput.get_element_type(), targetInputStaticShapes[0], 256, 0, 1);
+            } else if (funcInput.get_node()->get_friendly_name() == "indices") {
+                tensor = ov::test::utils::create_and_fill_tensor(
+                        funcInput.get_element_type(), targetInputStaticShapes[1], axisDim * 2, -axisDim, 1);
+            } else if (funcInput.get_node()->get_friendly_name() == "axis") {
+                tensor = ov::test::utils::create_and_fill_tensor(funcInput.get_element_type(), {1}, 1, axis, 1);
+            }
+            inputs.insert({funcInput.get_node_shared_ptr(), tensor});
+        }
+    }
+
+    int64_t axis = 0;
+};
+
+TEST_P(GatherLayerTestCPU, CompareWithRefs) {
+    SKIP_IF_CURRENT_TEST_IS_DISABLED()
+
+    run();
+    CheckPluginRelatedResults(executableNetwork, "Gather");
+}
+
+namespace {
+const std::vector<ElementType> netPrecisions = {
+        ElementType::i32, // TODO: remove
```

Comment:
Does this TODO comment still valid? Could you please elaborate why we have to remove this precision? And if so, why haven't we done it yet?

Reply:
FP32 is enough to represent 32bit data. I've added I32 for easy debugging.
Removed.